charts/external-dns/templates/deployment.yaml

```
@@ -132,6 +134,55 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- if eq $providerName "webhook" }}
+        {{- with .Values.provider.webhook }}
+        - name: webhook
+          {{- with .image }}
+          {{- if or (empty .repository) (empty .version) }}
+            {{- fail "ERROR: webhook provider needs an image repository and a version" }}
+          {{- end }}
+          image: {{ .repository }}:{{- .version }}
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          {{- end }}
+          {{- with .env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- with .args }}
+          args:
+            - {{ toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: webhook-http
+              protocol: TCP
+              containerPort: 8888
+            - name: webhook-metrics
```

```suggestion
            - name: http-webhook-metrics
```
This might be too long, but it would be the idiomatic pattern. I'm not sure if anything still relies on this convention but I generally follow it. If it's too long maybe the ports should use `wh` instead of `webhook`.